content/en/account_management/saml/troubleshooting.md

```diff
@@ -0,0 +1,82 @@
+---
+title: SAML Troubleshooting
+kind: documentation
+description: Troubleshoot SAML issues for your Datadog account
+further_reading:
+- link: "https://www.samltool.com/online_tools.php"
+  tag: "Developer Tools"
+  text: "Identify your assertions with developer SAML tools"
+
+---
+
+## Overview
+
+This page provides troubleshooting instructions for common errors during Security Assertion Markup Language (SAML) authentication.
+
+## IdP metadata file errors
+
+If you are having trouble with updating your IdP metadata file, it is recommended to verify that the metadata file you are trying to upload is valid.
+
+To validate your metadata file:
+
+1. Choose a SAML validation tool, such as the [SAML developer tool][1] by OneLogin.
+2. Paste your metadata into the XML field and select **Metadata** in the XSD (schema file) field.
+3. Click **Validate XML With the XSD Schema**.
+
+## Roles errors
+
+When mappings are enabled, users logging in with SAML to a Datadog account are stripped of their current roles and reassigned to new roles based on the details in their SAML assertion passed on from your Identity Provider, and the mappings set within Mappings settings.
+
+Users who log in with SAML and do not have the values that map to a Datadog role are stripped of all roles and will not be allowed to log in.
+
+If you have group mappings set and are not able to see your roles, you group mappings in the Datadog application may appear differently in your IdP. To verify:
+
+1. Retrieve your IdP's SAML assertion for your account. Use browser tooling, such as [extensions][6], to retrieve your SAML assertion.
+2. Navigate to the Team page in the bottom left corner of Datadog.
+3. Select the **Mappings** tab.
+4. Compare the attributes provided by your IdP in your SAML assertion to the attributes set in the **Mappings** tab.
+5. Resolve any discrepencies in either the Datadog Mappings settings, or within your IdP settings.
+
+If you are having trouble logging in because of a role-based error, contact your Administrator to complete the troubleshooting steps above.
+
+**Notes**: 
+
+- Each IdP provides different types of attributes, and different ways to set attributes. For example, Azure uses [object IDs][7] for their attribute, or if you're using Okta, you must set attributes in [Okta settings][8]. Reference your IdP's attribute documentation for information.
+
+- When you disable Mappings, users are allowed to login with SAML and have the same roles they are assigned to—even if the group membership changed in your IdP.
+
+## Identity provider errors
+
```

Comment:
Can we be more explicit here by saying something like: "If you encounter an error that comes from your idp" rather than errors that are Datadog errors ie:`Arf.`

Reply:
Definitely—and are there additional IdPs we can add to the list?

Reply:
All of the idps we have documentation for would be a good start. You can see them under account management: https://docs.datadoghq.com/account_management/saml/